docs/source/toolkit/testing.rst

```diff
@@ -0,0 +1,213 @@
+Provider's authentication. An example of how to create a :code:`client` fixture
+which can be used to make unauthenticated HTTP requests against the Action
+Provider is shown below:
+
+.. code-block:: python
+
+    @pytest.fixture(scope="module")
+    def client(flask_helpers_noauth):
+        app = create_app()
+        app.config["TESTING"] = True
+        yield app.test_client()
+
+Once composed like this, you can use the :code:`client` fixture in your tests to
+receive and use a Flask *test_client* to make requests against your Action
+Provider:
+
+.. code-block:: python
+
+    def test_introspection_endpoint(client):
+        response = client.get("/")
+        assert response.status_code == 200
+
+.. note:: 
+    
+    The :code:`flask_helpers_noauth` fixture will patch the TokenChecker in a
+    global scope during testing, meaning that any other Action Providers that
+    are themselves built using the Flask API Helpers will also have their
+    TokenChecker's patched. This may lead to unintended issues if testing
+    multiple Action Providers in the same pytest test session. If this is your
+    case, we highly recommend isolating your Action Provider tests.
+
+
+Mocks
+^^^^^
+
+The toolkit provides various `mocks
+<https://docs.python.org/3/library/unittest.mock.html#the-mock-class>`_ which
+can be used individually to stub out your Action Provider's authentication. You
+should use these directly if you are writing an Action Provider using a
+non-Flask framework or if you've decided not to use the built in Flask helpers.
+
+.. note::
+
+    This toolkit uses these mocks within the
+    :code:`globus_action_provider_tools.testing.fixtures` module.  
+
+
+.. _mock-authstate:
+
+Mock AuthState
+--------------
+
+An *AuthState* represents a requestor's authentication status and Globus Auth
+information. Every request should have its token validated via the
+*TokenChecker*'s :code:`check_token` method, which in turns generates an
+*AuthState* object.
+
+During testing, it is desirable to not need to provide valid tokens with every
+request. Use the :code:`mock_authstate` mock to generate a stubbed out
+*AuthState* object that won't validate requestor properties against Globus Auth.
+This is most useful when used in a patch as the return value for the
+*TokenChecker*'s :code:`check_token` method:
+
+.. code-block:: python
+
+    from unittest import mock     
+
+    import pytest
+    from globus_action_provider_tools.testing.mocks import mock_authstate
+
+    ...
+
+    @pytest.fixture
+    def client():
+        with mock.patch(
+            "globus_action_provider_tools.authentication.TokenChecker.check_token",
+            return_value=mock_authstate(),
+        ):
+            yield app.test_client()
+
+The example above creates a fixture which can be used to create a client that
+can make unauthenticated HTTP requests against an Action Provider. 
+
+
+Mock TokenChecker
```

Comment:
This seems like a really good approach for non-flask apps? Should we push this description forward so people get to it before the other approach?

Reply:
I haven't had the chance to test this on a non-Flask app to see how we could just mock the `TokenChecker` import so I'm not sure how/if it'll work. I think mocking the import would work but I'm not quite ready to claim that it's possible or that it's a good idea.

For now I've added it as a piece that can be used and outlined a particular way of using it that I know works (via the provided fixtures) we start experimenting with other frameworks or want to revisit how framework agnostic our tools are, this would be something to work on pushing forward.